src/drivers/CgreenDriver.cpp

```diff
@@ -0,0 +1,159 @@
+#include <cucumber-cpp/internal/drivers/CgreenDriver.hpp>
+
+#include <boost/bind.hpp>
+#include <boost/function.hpp>
+
+using namespace cgreen;
+
+namespace cucumber {
+namespace internal {
+
+
+bool CgreenStep::initialized = false;
+
+static const char* const blacklist[] = {
+        "Running",
+        "Completed",
+        "Failure",
+        "currentTest",
+        "\n",
+        "/"
+};
+
+class CukeCgreenInterceptor {
```

Comment:
@muggenhor got it. Well I took a stab at implementing the lazy evaluate singleton you requested without the unique_ptr. I'm not dynamically allocating memory for the singleton itself though there are fields in it that are allocated and freed in its destructor. My assumption is that that memory is being successfully managed since the Valgrind tests pass. Let me know what your thoughts are.